h/accounts/views.py

```diff
@@ -132,7 +132,24 @@ def _logout(self):
 class AjaxAuthController(AuthController):
     def login(self):
         try:
-            appstruct = self.form.validate(self.request.json_body.items())
+            json_body = self.request.json_body
+        except ValueError as err:
+            self.request.status_code = 400
+            error = 'Could not parse request body as JSON: ' + err.message
+            return ajax_payload(
+                self.request, {'status': 'failure', 'errors': error})
+
+        if not isinstance(json_body, dict):
+            self.request.status_code = 400
+            error = 'Request JSON body must have a top-level object'
+            return ajax_payload(
+                self.request, {'status': 'failure', 'errors': error})
+
+        json_body['username'] = unicode(json_body.get('username') or '')
```

Comment:
What are these lines supposed to be doing? Strings from JSON are unicode already, and missing form fields should be handled by the validator, not here.